src/sentry/grouping/component.py

```diff
@@ -37,44 +41,35 @@ class GroupingComponent:
 
     def __init__(
         self,
-        id,
-        hint=None,
-        contributes=None,
-        values=None,
-        variant_provider=False,
-        tree_label=None,
-        is_prefix_frame=None,
-        is_sentinel_frame=None,
+        id: str,
+        hint: str | None = None,
+        contributes: bool | None = False,
+        values: Sequence[str | GroupingComponent] | None = None,
+        variant_provider: bool = False,
+        tree_label: str | None = None,
+        is_prefix_frame: bool | None = None,
+        is_sentinel_frame: bool | None = None,
     ):
         self.id = id
 
         # Default values
-        self.hint = DEFAULT_HINTS.get(id)
-        self.contributes = None
+        self.hint = hint
+        self.contributes = contributes
         self.variant_provider = variant_provider
-        self.values = []
-        self.tree_label = None
-        self.is_prefix_frame = False
-        self.is_sentinel_frame = False
-
-        self.update(
```

Comment:
I do not know why we would call `self.update` instead of initializing the lines above.